javascript/ql/lib/semmle/javascript/frameworks/Express.qll

```diff
@@ -677,7 +677,21 @@ module Express {
 
     RequestInputAccess() {
       kind = "parameter" and
-      this = [queryRef(request), paramsRef(request)].getAPropertyRead()
+      (
+        // `req.query` / `req.params`.
+        // These are objects, so we prefer to use a property read if possible, otherwise we fall back to the object itself.
+        (
+          if exists(queryRef(request).getAPropertyRead())
+          then this = queryRef(request).getAPropertyRead()
+          else this = queryRef(request)
```

Comment:
```suggestion
          else this = request.ref().getAPropertyRead("query")
```
and similar for params (feel free to refactor to eliminate the duplication with `queryRef`).

Treating the result of the type-tracking predicate as a source can lead to redundant sources:
```js
foo(req.query); // <-- 'req.query' is the correct source
function foo(query) {} // <-- 'query' becomes a redundant source
```